### PR TITLE
Add fuerte testsuite

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

- Add fuerte tests from other repo
- Add more tests for VST / HTTP2 and fix the timeout behaviour
- Also includes the changes of PR https://github.com/arangodb/arangodb/pull/12549
- Also tests short keep alive timeouts, i.e. **BTS-173**

- [x] :pizza: New feature 
- [x] :hammer: Refactoring 

#### Backports:

- [x] No backports required
- [ ] Backports required for: *(Please specify versions)*

### Testing & Verification

- [x] This change is already covered by existing tests, such as  **shell_client** testsuite
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new C++ **Unit Tests**
  - [x] Added new **integration tests** fuerte test suite

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11710/

### CODE REVIEW

File: js/client/modules/@arangodb/testsuites/fuerte.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  ];
+  if (options.hasOwnProperty('testCase') && (typeof (options.testCase) !== 'undefined')) {
+    argv.push('--gtest_filter=' + options.testCase);
+  } /*else {
+    argv.push('--gtest_filter=-*_LongRunning');
+    let greylist =  readGreylist();
+    greylist.forEach(function(greyItem) {
+      argv.push('--gtest_filter=-'+greyItem);
+    });
+  }*/
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testsuites/fuerte.js:144]:
```suggestion
  }
```

File: js/client/modules/@arangodb/testsuites/fuerte.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  }
+  results = getGTestResults(testResultJsonFile, results);
+
+
+  print('Shutting down...');
+
+  // if (result[te].status === false) {
+  //   options.cleanup = false;
+  // }
+
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testsuites/fuerte.js:168]:
```suggestion
```

File: js/client/modules/@arangodb/testsuites/fuerte.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  results['shutdown'] = pu.shutdownInstance(instanceInfo, options);
+
+  print('done.');
+
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testsuites/fuerte.js:172]:
```suggestion
```

File: 3rdParty/fuerte/src/VstConnection.cpp [hunk 4]
```diff
@@ -373,104 +368,32 @@ std::unique_ptr<fu::Response> VstConnection<ST>::createResponse(
...
-    for (auto const& pair : _messages) {
-      pair.second->invokeOnError(err);
-    }
-    _messages.clear();
-    _numMessages.store(0);
-  }
-}
-
-/// Set timeout accordingly
+/// abort ongoing / unfinished requests expiring before given timpoint
```
Comment (jsteemann) [inline 3rdParty/fuerte/src/VstConnection.cpp:371]:
```suggestion
/// abort ongoing / unfinished requests expiring before given timepoint
```

**Comment by jsteemann** [comment]
I think this does not run the fuerte tests right now.
We need to add them at least here:
* https://github.com/arangodb/oskar/blob/master/scripts/runFullTests.fish#L76
* https://github.com/arangodb/oskar/blob/master/scripts/runFullTests.ps1#L79
* https://github.com/arangodb/oskar/blob/master/scripts/runTests.fish#L31
* https://github.com/arangodb/oskar/blob/master/scripts/runTests.ps1#L64
and then put them into the blocklist for 3.5, 3.6 and 3.7.

**Comment by jsteemann** [comment]
Oskar PR: https://github.com/arangodb/oskar/pull/263

**Comment by jsteemann** [comment]
Tests run with Oskar PR to enable fuerte testsuite: http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11712/

**Comment by graetzer** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/11728/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11765/

**Comment by jsteemann** [comment]
Running it on Oskar branch that enables fuerte tests:
http://172.16.10.101:8080/job/arangodb-matrix-pr/11778/

File: tests/Fuerte/test_10000_writes.cpp [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+ private:
+  f::EventLoopService _eventLoopService;
+  std::string _server;
+  std::string _port;
+};
+
+namespace fu = ::arangodb::fuerte;
+
+/*
+TEST_F(Connection100kWritesF, Writes100k){
```
Comment (danielhlarkin) [inline tests/Fuerte/test_10000_writes.cpp:106]: Why is this test disabled?

Comment (graetzer) [reply tests/Fuerte/test_10000_writes.cpp:106]: This test file does not build properly anymore, but could possibly deleted

**Comment by graetzer** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/11780/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**